eclipse-scout-core/src/desktop/outline/OutlineAdapter.ts

```diff
@@ -251,72 +260,115 @@ export class OutlineAdapter extends TreeAdapter {
     this.modelAdapter._linkNodeWithRowLater(page);
   }
 
-  protected static override _createTreeNodeRemote(this: Tree & { modelAdapter: TreeAdapter; _createTreeNodeOrig }, nodeModel: TreeNodeModel) {
+  protected static override _createTreeNodeRemote(this: Outline & { modelAdapter: OutlineAdapter; _createTreeNodeOrig }, pageModel: PageModel) {
     // nodeType is only used for Scout Classic pages
-    if (!this.modelAdapter || !nodeModel?.nodeType) {
-      return this._createTreeNodeOrig(nodeModel);
+    if (!this.modelAdapter || !pageModel?.nodeType) {
+      return this._createTreeNodeOrig(pageModel);
     }
 
-    nodeModel = (this.modelAdapter as OutlineAdapter)._initNodeModel(nodeModel);
-
-    if (nodeModel.nodeType === 'jsPage') {
-      if (!nodeModel.jsPageObjectType) {
-        throw new Error('jsPageObjectType not set');
+    pageModel = this.modelAdapter._initNodeModel(pageModel);
+    if (pageModel.nodeType === 'jsPage') {
+      try {
+        return this.modelAdapter._createJsPage(pageModel, this._createTreeNodeOrig.bind(this));
```

Comment:
Why is the real page returned as an instance, but the error page only as a model? Can we also create an instance for the error page?

Reply:
Because createJsPage does an assertion for classId at the end for which it needs to be a real page. The broken page will be created "automatically" at the end of the createTreeNodeRemote function, so it is not necessary